tests/functional_test_base.go

```diff
@@ -415,3 +418,8 @@ func (s *FunctionalTestBase) registerArchivalNamespace(archivalNamespace string)
 	)
 	return err
 }
+
+// MatchingServiceClient returns the matching service client setup for this functional test.
```

Comment:
Note that this client doesn't do routing, so it doesn't work for more than one matching node. There's a comment in `temporalImpl.GetMatchingClient` to this effect but it should probably be copied to all the accessors up the call path (or else fix the issue).

But also, none of this will work with external servers since we don't have access to membership... I think we may have passed the point where functional tests can still work with external servers.

Maybe what we should do is: `t.Skip()` if this is nil? That way the tests that need only frontend can run against external ones, and tests that need access to matching (or history) will get automatically skipped?

(And also fix it to grab the actual routing client from the client bean from fx after it's constructed)